### ISSUE TITLE
Recording truncated

### ISSUE THREAD

**Issue by Mte90** [lagmoellertim/unsilence#108]
**Describe the bug**
Since 2 weeks unsilence is not working anymore to me.

**To Reproduce**

```unsilence $1 pulito-$1 -stt 0.4 -ao -sl '-18' -t 4```

I always used in this way to clean my weekly podcast.
As example if I run this on the mp3 of this episode https://funkwhale.it/library/tracks/2065/ (that I cleaned manually after the issue to publish it)
![Screenshot_20210914_154002](https://user-images.githubusercontent.com/403283/133268488-a7f9de5c-ac7d-4ce2-b748-80a76cb6ed63.png)
Unsilence detect the silence, and generate a cleaned file. With the original before the manual clean it was truncating the audio.

So if you download https://funkwhale.it/library/tracks/2057/ the previous episode that wasn't cleaned manually the problem is happening
![Screenshot_20210914_154434](https://user-images.githubusercontent.com/403283/133268987-9efa83e3-2a6f-4432-b9de-516da5dcfe11.png)

It is again everything working but the audio file generated:
![immagine](https://user-images.githubusercontent.com/403283/133269096-4605014b-1142-41f9-8aab-69c521fd52c1.png)
Instead is just 2~ minutes

**Desktop (please complete the following information):**
 - OS + version: Debian sid
 - Python version 3.9.7
 - FFMPEG version 4.4-6

**Comment by Mte90** [comment]
It is happened also with this recording https://www.spreaker.com/user/mte90/50special
![immagine](https://user-images.githubusercontent.com/403283/144058113-cbb04ba8-a91a-499b-9135-5daa49b64865.png)

At the end there are just 8 minutes but the recording so more long

**Comment by Mte90** [comment]
So another info I had issue with the podcast recorded today.
The recording was failing again on 31 minutes there were just the first 13 minutes.
So I started editing the original audio and removing some silences (just 6 seconds) and now unsilence detected everything and generated a right file.

**Comment by lagmoellertim** [comment]
There seems to be a bug in the Interval Combination Method, since the only parameter that fails is silence_level, which only influences the Interval Size and the FFMPEG Command directly. Currently, I am sadly a little short on time, but this is on my todo to check what is wrong.

**Comment by Mte90** [comment]
Any updates?

**Comment by Mte90** [comment]
Any updates for this? 

This tool worked very good as now and migrate to another one...

**Comment by Mte90** [comment]
@lagmoellertim just a ping

**Comment by Garbaz** [comment]
![image](https://user-images.githubusercontent.com/13696798/197990154-a5b52ca2-6e0c-4b78-a43e-46a8e9f1ecb1.png)

I have the same issue. If I try to unsilence a certain ~47min file, which according to the output of unsilence should result in a ~42min file, I instead just get the first ~6min as the result.

#### Observations:

* Adding `drop-corrupted-intervals` doesn't change anything.
* Changing the `silence-level` to `-40`, which should result in a file almost identical to the original, does give me a bigger file, but it still is missing a part at the end, being only ~40min long.
* **Changing the `silent-speed` to `6` does result in a complete file.**

_Additional note:_ I tried a few different files, and the "Combining Intervals" progress bar always stays at 0%, and the tool pretty much instantly finishes the moment the "Rendering Intervals" progress bar is at 100%, irrespective of whether the tool works correctly or doesn't. So while this is curious, it doesn't appear to have anything to do with this bug.

**Comment by Mte90** [comment]
@lagmoellertim can you help us? If I get some hints I can try to do myself the patch to the project.

**Comment by Garbaz** [comment]
### Further Observations:

#### Does work with MP3 in / WAV out

![image](https://user-images.githubusercontent.com/13696798/198001293-9515dbf3-bf8f-4449-9353-1424a020a7d2.png)

Running the tool on exactly the same input file, but giving the output file as a WAV instead of MP3 does result in a complete file. However, the "Combining Intervals" bar still doesn't move.

#### Doesn't work with WAV in / MP3 out

Results in a ~16min file.

**Comment by Garbaz** [comment]
### Investigation

When I run the `ffmpeg` command generated in `MediaRenderer.__concat_intervals` manually:

```bash
ffmpeg -f concat -safe 0 -i conc.txt -c copy -y -loglevel verbose test.mp3
```

where the files listed in `conc.txt` are simply stolen from the `.tmp/*` folder generated by unsilence, I get an interesting output, that being that the `ffmpeg` runs fine until it hits `out_155.mp3`, where it complains:

![image](https://user-images.githubusercontent.com/13696798/198005128-b0602c40-a30b-4202-8808-c2e7401356d7.png)

And voila, the file I get out is the ~6min one I was getting earlier.

If I try to open `out_155.mp3` with `mpv`, it just spits `Failed to recognize file format.` back at me. However, the same happens if I open some of the other files. If I just let `mpv` play through all `out_*.mp3` files:

```bash
find . -name "out_*" | sort -V | xargs mpv
```

I get every few files a `Failed to recognize file format.` and that file is skipped in playback.

![image](https://user-images.githubusercontent.com/13696798/198010107-445cec0f-66e8-4d3b-a6b7-418125d2afde.png)

However, except for `out_155.mp3`, all these files that aren't recognized by mpv are 1140 bytes in size and only raise a warning from ffmpeg: "Format mp3 detected only with low score of 1, misdetection possible!". `out_155.mp3` itself is 1976 bytes in size.

If I insert a `print(console_output)` in `RenderInvervalThread.__render_interval` and look at what the ffmpeg command there, which is generating the file, has to say about `out_155.mp3`, then it doesn't appear to be different to any other intervals, however what is noticeable is that the interval is, after speedup, incredibly short (`time=00:00:00.07`). Though the other intervals (they 1140 byte files) all are even shorter (`time=00:00:00.02`).

### Conclusion

Given all the above, it appears to me that there just seems to be some error in ffmpeg which messes up these super short MP3 files. And I'm not gonna start digging around in the ffmpeg source to figure out why.

So the best I can come up with to resolve the problem, is to ensure that we don't include files that ffmpeg doesn't like in the final concatenation. Ideally we could just tell ffmpeg's concat script to skip any files it can't read, but as far as I can tell, there doesn't appear to be an option for that. However, we can check with `ffprobe` whether ffmpeg does like a certain file.

I have implemented this and created a pull request (#140).

**Comment by lagmoellertim** [comment]
If this only happens if the resulting files are very short, couldn't a flag that specifies the minimum length of an interval after speedup be used? At least this would not increase the total rendering time like checking each interval with `ffprobe` would do.

**Comment by Mte90** [comment]
I have issues also with files of 40 minutes so and using it with the parameter in the first ticket comment (there are also some files that replicate the issue).

**Comment by Garbaz** [comment]
> I have issues also with files of 40 minutes so and using it with the parameter in the first ticket comment (there are also some files that replicate the issue).

The length of the total file is not what they are referring to (my problematic file also is around 40mins long), but the length of the individual interval files the program generates as an intermediate step.

**Comment by lagmoellertim** [comment]
Since I merged PR #140, this problem should hopefully be fixed now.

**Comment by Mte90** [comment]
I am testing with https://funkwhale.it/library/tracks/2057/

Report this
![Screenshot_20221103_122800](https://user-images.githubusercontent.com/403283/199709552-0e0d2fb6-b29d-4eeb-8a3f-d71a83b39e96.png)

But in reality the audio is 23:05 instead of 22:11 in the output file.
